R/workflow_map.R

```diff
@@ -208,13 +209,52 @@ allowed_fn <-
   tibble::tibble(
     func = c(
       "tune_grid", "tune_bayes", "fit_resamples", "tune_race_anova",
-      "tune_race_win_loss", "tune_sim_anneal"
+      "tune_race_win_loss", "tune_sim_anneal", "tune_cluster"
     ),
-    pkg = c(rep("tune", 3), rep("finetune", 3))
+    pkg = c(rep("tune", 3), rep("finetune", 3), "tidyclust")
   )
 allowed_fn_list <- paste0("'", allowed_fn$func, "'", collapse = ", ")
 # nocov end
 
+# ---------------------------------------------
+check_object_fn <- function(object, fn, call = rlang::caller_env()) {
+   wf_specs <- purrr::map(object$wflow_id, ~extract_spec_parsnip(object, id = .x))
+   is_cluster_spec <- purrr::map_lgl(wf_specs, inherits, "cluster_spec")
+
+   if (identical(fn, "tune_cluster")) {
+      if (!all(is_cluster_spec)) {
+         cli::cli_abort(
+            "To tune with {.fn tune_cluster}, each workflow's model \\
+            specification must inherit from {.cls cluster_spec}, but \\
+            {.var {object$wflow_id[!is_cluster_spec]}} {?does/do} not.",
+            call = call
+         )
+      }
+      return(invisible())
+   }
+
+   is_model_spec <- purrr::map_lgl(wf_specs, inherits, "model_spec")
+
+   msg <-
+    "To tune with {.fn {fn}}, each workflow's model \\
+     specification must inherit from {.cls model_spec}, but \\
+     {.var {object$wflow_id[!is_model_spec]}} {?does/do} not."
+
+   if (any(is_cluster_spec)) {
+      msg <- c(
+         msg,
+         "i" = "{cli::qty(object$wflow_id[is_cluster_spec])} The workflow{?/s} \\
+                {.var {object$wflow_id[is_cluster_spec]}} {?is a /are} cluster
+                specification{?/s}. Did you intend to set `fn = 'tune_cluster'`?"
```

Comment:
This changes 2 things:
- there was a missing `\\`
- some of the lines were too long

```suggestion
         "i" = "{cli::qty(object$wflow_id[is_cluster_spec])} \\
                The workflow{?/s} {.var {object$wflow_id[is_cluster_spec]}} \\
                {?is a /are} cluster specification{?/s}. Did you intend to \\
                set `fn = 'tune_cluster'`?"
```